### PR TITLE
JWT Improvements

### PR DESCRIPTION
**Pull request by mattupstate**
Please not that I've removed the notion of a user and went with the notion of identity because JWT's are not necessarily always for the common concept of a user, that being API interactions initiated by human interaction.

### CODE REVIEW

File: flask_jwt/__init__.py [hunk 1]
```diff
@@ -71,53 +30,58 @@ def _default_response_handler(payload):
...
-    realm = realm or current_app.config['JWT_DEFAULT_REALM']
-    auth = request.headers.get('Authorization', None)
+def _default_jwt_decode_handler(token):
+    secret = current_app.config['JWT_SECRET_KEY']
+    algorithm = current_app.config['JWT_ALGORITHM']
+
+    verify_claims = current_app.config['JWT_VERIFY_CLAIMS']
+    required_claims = current_app.config['JWT_REQUIRED_CLAIMS']
+
+    options = {
```
Comment (f0t0n) [inline flask_jwt/__init__.py]: Will be always equal to `None`.

Instead should look like

```
options = {
    'verify_{}'.format(claim): True
    for claim in verify_claims
}
options.update({
    'require_{}'.format(claim): True
    for claim in required_claims
})
```

Or somehow like this:

```
from itertools import chain

verify_claims = map(
    lambda x: ('verify_{}'.format(x), True),
    current_app.config['JWT_VERIFY_CLAIMS']
)
required_claims = map(
    lambda x: ('require_{}'.format(x), True),
    current_app.config['JWT_REQUIRED_CLAIMS']
)
options = dict(chain(verify_claims, required_claims))

```

Comment (mattupstate) [inline flask_jwt/__init__.py]: Good catch

**Comment by hairychris** [comment]
PyJWT looks like a sensible choice, much more actively maintained and they responded to the choosing your own algorithm security issue quickly back in March. We just need to make sure to whitelist algorithms to avoid any issues.

Changing user to identity is a good change too, more flexible.

I'll be using this over the weekend so will add anything else as it comes up.

**Comment by f0t0n** [comment]
So since it's not just a matter of refactoring but also of security measure (`PyJWT` vs `itsdangerous`) could we speed-up the new version release based on current PR?

**Comment by Xk0nSid** [comment]
@mattupstate When is this update to be merged with master? I'm planning to use this in a project. Would prefer `PyJWT` version instead of `itsdangerous` version.

**Comment by mattupstate** [comment]
I'll do my best to address this, but I can't make a promise as to when because I'm getting married next week. So hopefully you can imagine how busy I've been recently.

**Comment by Xk0nSid** [comment]
@mattupstate First of all congratulations and no worries, I can still work with the `itsdangerous` version.

**Pull request closed without merging**